sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -341,3 +341,53 @@ case class MultiplyDTInterval(
 
   override def toString: String = s"($left * $right)"
 }
+
+// Divide an year-month interval by a numeric
+case class DivideYMInterval(
+    interval: Expression,
+    num: Expression)
+  extends BinaryExpression with ImplicitCastInputTypes with NullIntolerant with Serializable {
+  override def left: Expression = interval
+  override def right: Expression = num
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(YearMonthIntervalType, NumericType)
+  override def dataType: DataType = YearMonthIntervalType
+
+  @transient
+  private lazy val evalFunc: (Int, Any) => Any = right.dataType match {
+    case LongType => (months: Int, num) =>
+      LongMath.divide(months, num.asInstanceOf[Long], RoundingMode.HALF_UP).toInt
```

Comment:
`months` is int, the result of `/` must fit to int too. So, `toInt` should be safe.

Reply:
It's better to put it as a code comment. We can fix it in the next PR.

Reply:
I added comments in https://github.com/apache/spark/pull/31972